elements/lisk-blocks/src/data_access/storage_access.ts

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright © 2019 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import {
+	TransactionJSON,
+} from '@liskhq/lisk-transactions';
+
+import {
+	Account,
+	BlockHeader,
+	BlockJSON,
+	RoundDelegates,
+	Storage,
+	StorageTransaction,
+	TempBlock,
+} from '../types';
+
+export class StorageAccess {
+	private readonly _storage: Storage;
+
+	public constructor(storage: Storage) {
+		this._storage = storage;
+	}
+
+	public async getBlockHeadersById(
+		arrayOfBlockIds: string[],
+		tx?: StorageTransaction,
+	): Promise<BlockHeader[]> {
+		const blocks = await this._storage.entities.Block.get(
+			{ id_in: arrayOfBlockIds },
+			{},
+			tx,
+		);
+
+		return blocks;
+	}
+
+	public async getBlockHeadersByHeight(
+		fromHeight: number,
+		toHeight: number,
+		tx?: StorageTransaction,
+	): Promise<BlockHeader[]> {
+		const blocks = await this._storage.entities.Block.get(
+			{ height_gte: fromHeight, height_lte: toHeight },
+			{},
+			tx,
+		);
+
+		return blocks;
+	}
+
+	public async getBlockHeadersWithHeights(
+		heightList: ReadonlyArray<number>,
+	): Promise<BlockHeader[]> {
+		const blocks = await this._storage.entities.Block.get(
+			{
+				height_in: heightList,
+			},
+			{
+				sort: 'height:asc',
+				limit: heightList.length,
+			},
+		);
+
+		return blocks;
+	}
+
+	public async getBlockHeadersInRounds(query: {
```

Comment:
```suggestion
	public async getBlockHeadersWithInterval(query: {
```
and it's better to consistently accepts the input rather than object, and it shouldn't have any `rounds` mentioned in the blocks